### PR TITLE
[SW-1977] Create parameterless methods getOrCreate as we don't need to pass spark anymore

### PR DESCRIPTION
**Pull request by jakubhava**
This change allows to call getOrCreate in R/Py/Scala without the spark/sc argument. Updated all references in the codebase. The next change is the deprecation of the old methods - https://0xdata.atlassian.net/browse/SW-1970

### CODE REVIEW

File: doc/src/site/sphinx/tutorials/change_log_location.rst [hunk 1]
```diff
@@ -21,14 +21,14 @@ In Scala:
 .. code:: scala
 
     val conf = new H2OConf(spark).setH2OClientLogDir("log_location")
-    val hc = H2OContext.getOrCreate(spark, conf
+    val hc = H2OContext.getOrCreate(conf)
```
Comment (mn-mikke) [inline doc/src/site/sphinx/tutorials/change_log_location.rst]: :+1:

File: doc/src/site/sphinx/tutorials/secured_flow.rst [hunk 2]
```diff
@@ -77,7 +77,7 @@ In order to use https correctly, the following two options need to be specified:
 
             from pysparkling import *
             conf = H2OConf(spark).setJks("/path/to/keystore").setJksPass("password)
-            hc = H2OContext.getOrCreate(spark, conf, verify_ssl_certificates=False)
+            hc = H2OContext.getOrCreate(onf, verify_ssl_certificates=False)
```
Comment (mn-mikke) [inline doc/src/site/sphinx/tutorials/secured_flow.rst]: typo: `onf` -> `conf`

File: py/src/ai/h2o/sparkling/H2OContext.py [hunk 3]
```diff
@@ -218,7 +200,8 @@ def asSparkFrame(self, h2oFrame, copyMetadata=True):
         if isinstance(h2oFrame, H2OFrame):
             frame_id = h2oFrame.frame_id
             jdf = self._jhc.asDataFrame(frame_id, copyMetadata)
-            df = DataFrame(jdf, self._sql_context)
+            sqlContext = SparkSession.builder.getOrCreate()._wrapped
```
Comment (mn-mikke) [inline py/src/ai/h2o/sparkling/H2OContext.py]: nit: The varialble is named `sqlContext` but isn't it a spark session?

File: doc/src/site/sphinx/tutorials/secured_flow.rst (hunk 2 above)
Comment (jakubhava) [reply doc/src/site/sphinx/tutorials/secured_flow.rst]: 👍

File: py/src/ai/h2o/sparkling/H2OContext.py (hunk 3 above)
Comment (jakubhava) [reply py/src/ai/h2o/sparkling/H2OContext.py]: https://github.com/apache/spark/blob/bcce1b10408b5c516c936e8770a598ef415d71c3/python/pyspark/sql/session.py#L231 Should be ``SqlContext``

**Pull request merged**